src/devtools/DevToolsViewProvider.ts

```diff
@@ -20,16 +49,40 @@ export class DevToolsViewProvider implements vscode.WebviewViewProvider {
       panel.webview,
       this._extensionUri,
     );
-
-    panel.webview.onDidReceiveMessage((data) => {
-      devtoolsEvents.emit("fromDevTools", data);
-    });
+    const panelDisposables: vscode.Disposable[] = [];
+    panel.webview.onDidReceiveMessage(
+      (data) => {
+        if (data.source === "apollo-client-devtools") {
+          devtoolsEvents.emit("fromDevTools", data);
+        }
+        if (data.source === "vscode-panel") {
+          if (data.type === "mounted") {
```

Comment:
```suggestion
        if (data.source === "vscode-panel" && data.type === "mounted") {
```

You should be able to combine these conditionals.

Reply:
I assume that inside of `data.source` we'll see more reactions to other `data.type` in the future, which is why I separated those from each other.